internal/pkg/rdr/rdr.go

```diff
@@ -220,7 +220,9 @@ func (r *Reader) nextBlock() (err error) {
 }
 
 func (r *Reader) modeHeader(dst []byte) (int, error) {
-	_, err := r._readHeader()
+	n, err := r._readHeader()
+
+	r.srcPos += int64(n)
```

Comment:
doing this before error check, is this correct?

Reply:
Yeah, in general supposed to accept bytes before processing error.  From io.Reader

<<
Callers should always process the n > 0 bytes returned before considering the error err. Doing so correctly handles I/O errors that happen after reading some bytes and also both of the allowed EOF behaviors.
>>

In practice it won't matter in this case  because the call will error out and the reader will be in an error state.